src/device/device-management.ts

```diff
@@ -33,29 +33,87 @@ export type AudioDeviceConstraints = {
   deviceId?: string;
 };
 
-export type VideoDeviceConstraints = {
-  deviceId?: ConstrainDOMString;
-  width?: ConstrainULong;
-  height?: ConstrainULong;
-  aspectRatio?: ConstrainDouble;
-  frameRate?: ConstrainDouble;
-  facingMode?: ConstrainDOMString;
+// Predefined encoder config mentioned similar to https://api-ref.agora.io/en/voice-sdk/web/4.x/globals.html#videoencoderconfigurationpreset
+export enum StaticVideoEncoderConfig {
+  Profile1080P = '1080P',
+}
+
+export type ConstrainLong = {
+  min?: number;
+  max?: number;
+  ideal?: number;
+  exact?: number;
+};
+
+// Predefined encoder config mentioned similar to https://api-ref.agora.io/en/voice-sdk/web/4.x/globals.html#videoencoderconfigurationpreset
+export type videoEncoderConfig = {
+  bitRateMin?: number;
+  bitrateMax?: number;
+  frameRate?: number | ConstrainLong;
+  height?: number | ConstrainLong;
+  width?: number | ConstrainLong;
+  aspectRatio?: number;
+};
+
+export type VideoConstraints = {
+  cameraDeviceId?: string;
+  faceingMode?: ConstrainULong;
+  optimizationMode?: ConstrainULong;
+  encoderConfig?: StaticVideoEncoderConfig | videoEncoderConfig;
```

Comment:
So fields were removed from this?  I think this was designed to mimic the underlying API.